src/inmanta/deploy/scheduler.py

```diff
@@ -45,6 +54,13 @@
 LOGGER = logging.getLogger(__name__)
 
 
+@dataclass
+class ResourceIntent:
+    state_version: int
+    resource: ResourceDetails
+    dependencies: dict[ResourceIdStr, const.ResourceState]
```

Comment:
Nice! One suggestion (optional). And a few naming suggestions to be consistent with the rest of the code:
```suggestion
@dataclass(frozen=True)
class ResourceIntent:
    model_version: int
    details: ResourceDetails
    dependencies: Mapping[ResourceIdStr, const.ResourceState]
```